examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```
@@ -0,0 +1,163 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ConfigurationStoreBackup
+{
+    public static class BackupAppConfigurationStore
+    {
+        private static readonly ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
+        private static readonly ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("SecondaryStoreEndpoint")), new ManagedIdentityCredential());
+        private static readonly QueueClient queueClient = new QueueClient(Environment.GetEnvironmentVariable("AzureWebJobsStorage"), "eventgridqueue");
+        private const int maxMessagesToRead = 32;
+
+        [FunctionName("BackupAppConfigurationStore")]
+        public static void Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+
+            if (primaryAppConfigClient == null || secondaryAppConfigClient == null)
+            {
+                log.LogError($"Could not connect to App Configuration Store. Please validate the store endpoints that you have provided.");
+                return;
+            }
+
+            // Block the thread so that another function instance cannot be triggered while this is still processing.
+            ProcessStorageQueueMessagesAsync(log).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        private static async Task ProcessStorageQueueMessagesAsync(ILogger log)
+        {
+            // Peek to see if there are events in the queue.
+            while ((await queueClient.PeekMessagesAsync().ConfigureAwait(false)).Value.Length > 0)
+            {
+                Response<QueueMessage[]> retrievedMessages = await queueClient.ReceiveMessagesAsync(maxMessagesToRead).ConfigureAwait(false);
```

What transient error this call can throw? We should swallow those errors that can be ignored. We certainly don't want the Function crash for those.

---

Reading messages (get and peek both) will not throw any transient errors. 

---
In reply to: [412706456](https://github.com/Azure/AppConfiguration/pull/311#discussion_r412706456) [](ancestors = 412706456)